src/Microsoft.Sbom.Api/Manifest/ManifestConfigHandlers/SPDX22ManifestConfigHandler.cs

```diff
@@ -1,68 +1,33 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using Microsoft.Sbom.Api.Manifest.Configuration;
-using Microsoft.Sbom.Api.Recorder;
 using Microsoft.Sbom.Common;
 using Microsoft.Sbom.Common.Config;
 using Microsoft.Sbom.Extensions;
+using Microsoft.Sbom.Extensions.Entities;
 using Constants = Microsoft.Sbom.Api.Utils.Constants;
 
 namespace Microsoft.Sbom.Api.Manifest.ManifestConfigHandlers;
 
 /// <summary>
 /// Provides the ManifestConfig for the SPDX 2.2 format.
 /// </summary>
-public class SPDX22ManifestConfigHandler : IManifestConfigHandler
+public class SPDX22ManifestConfigHandler : BaseManifestConfigHandler
 {
-    private readonly IMetadataBuilderFactory metadataBuilderFactory;
-    private readonly IConfiguration configuration;
-    private readonly IFileSystemUtils fileSystemUtils;
-
-    private string ManifestDirPath => configuration.ManifestDirPath?.Value;
-
-    // directory path for SPDX 2.2 is
-    // root/_manifest/spdx_2.2/
-    private string SbomDirPath => fileSystemUtils.JoinPaths(ManifestDirPath, $"{Constants.SPDX22ManifestInfo.Name.ToLower()}_{Constants.SPDX22ManifestInfo.Version.ToLower()}");
-
-    // sbom file path is manifest.spdx.json in the sbom directory.
-    private string SbomFilePath => fileSystemUtils.JoinPaths(SbomDirPath, $"manifest.{Constants.SPDX22ManifestInfo.Name.ToLower()}.json");
-
-    // sha file is sbom file + .sha256
-    private string ManifestJsonSha256FilePath => $"{SbomFilePath}.sha256";
-
-    // catalog file is always manifest.cat
-    private string CatalogFilePath => fileSystemUtils.JoinPaths(SbomDirPath, Constants.CatalogFileName);
-
-    // bsi.json file contains build session metadata and is always bsi.json
-    private string BsiJsonFilePath => fileSystemUtils.JoinPaths(SbomDirPath, Constants.BsiFileName);
-
-    private IMetadataBuilder MetadataBuilder => metadataBuilderFactory.Get(Constants.SPDX22ManifestInfo);
-
     public SPDX22ManifestConfigHandler(
         IConfiguration configuration,
         IFileSystemUtils fileSystemUtils,
         IMetadataBuilderFactory metadataBuilderFactory)
+        : base(configuration, fileSystemUtils, metadataBuilderFactory)
     {
-        this.metadataBuilderFactory = metadataBuilderFactory ?? throw new ArgumentException(nameof(metadataBuilderFactory));
-        this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
-        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
     }
 
-    public bool TryGetManifestConfig(out ISbomConfig sbomConfig)
+    /// <inheritdoc/>
+    protected override ManifestInfo ManifestInfo => Constants.SPDX22ManifestInfo;
+
+    public override bool TryGetManifestConfig(out ISbomConfig sbomConfig)
```

Comment:
The only difference in the implementations of `SPDX22ManifestConfigHandler.TryGetManifestConfig` and `SPDX30ManifestConfigHandler.TryGetManifestConfig` is the `ManifestInfo` you're looking for. You could move the common code into a single method:

```
protected bool TryGetManifestConfig(out ISbomConfig sbomConfig, ManifestInfo ManifestInfo)
```

Then the overridden versions both look the same because you've already overridden `ManifestInfo`:

```
public override bool TryGetManifestConfig(out ISbomConfig sbomConfig)
{
    return base.TryGetManifestConfig(sbomConfig, ManifestInfo;
}
```